handlers.rb

```diff
@@ -0,0 +1,116 @@
+module Handlers
+  def list_books
+    @my_books.each_with_index do |x, index|
+      puts "#{index}) Title: \"#{x.title}\", Author: #{x.author}"
+    end
+  end
+
+  def list_people
+    @people.each_with_index do |x, index|
+      puts "#{index}) [#{x.class.name}] Name: #{x.name}, ID: #{x.id}, Age: #{x.age}"
+    end
+  end
+
+  def action_list_books
+    p 'Book successfully created!'
+    list_books
+    p 'Press enter to continue...'
+    gets.chomp
+    run
+  end
+
+  def action_list_people
+    list_people
+    p 'Press enter to continue...!'
+    gets
+    run
+  end
+
+  def my_permission(my_char)
+    case my_char
+    when 'N'
+      false
+    when 'Y'
+      true
+    end
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, nil, specialization)
+    @people.push(teacher)
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = Student.new(age, name, my_permission(permission), nil)
+    @people.push(student)
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
+    puts 'Person created successfully!'
+    gets
+    run
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @my_books.push(new_book)
+    run
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by index'
+    list_books
+    book_index = gets.chomp
+    puts 'Select a person from the following list by index'
+    list_people
+    person_index = gets.chomp
+    print '\n Date(yyyy/mm/dd): '
+    rental_date = gets.chomp
+    new_rental = Rental.new(rental_date, @my_books[book_index.to_i], @people[person_index.to_i])
+    @my_rentals.push(new_rental)
+    puts 'Rental added successfully!'
+    run
+  end
+
+  def list_rental
+    me = nil
+    @people.each_with_index.map do |person, i|
+      puts "#{i}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, age: #{person.age}"
+    end
+    print 'Please, write the ID of the person: '
+    person_id = gets.chomp
+    @people.each do |x|
+      me = x if x.id == person_id.to_i
+    end
+    me.rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by Author: #{rental.book.author} "
+    end
+    puts 'Press any key to continue..'
+    run
+    puts 'Press any key to continue..'
+  end
+end
```

Comment:
- [ ] Kindly be noticed that there is no need to put these methods in a new file you have to put them in the `app.rb` file.
```ruby
require "./book.rb"
require "./student.rb"
require "./teacher.rb"
require "./rental.rb"

class App
  def initialize
    @books = []
    @people = []
    @rentals = []
  end
  
  # *Good example of Single responsability rule being followed
  def list_books
    @books.each do |book|
      puts book
    end
  end
  
  # *Good example of Single responsability rule being followed
  def list_people
    @people.each do |person|
      puts person
    end
  end
  
  # *Bad example of Single responsability rule
  # if you see this Remind them about the Best practices and Solid lesson: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid
  def create_person
    # *all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.
    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
    person_type = gets.chomp

    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end
    
    print "Age: "
    age = gets.chomp

    print "Name: "
    name = gets.chomp
    
    person =
      if person_type == "1"
        create_student(age, name)
      elsif person_type == "2"
        create_teacher(age, name)
      end

    @people << person
    puts "Person created successfully"
  end

  .
  .
  .

  private
  # *Bad example of Single responsability rule, this is also a
  # *Bad example of YAGNI and KISS as the contents fo this method could've been placed somewhere else
  # if you see this Remind them about the Best practices and Solid lesson: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid
  def create_student(age, name)
    # *all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.
    print "Has parent permission? [Y/N]: "
    parent_permission = gets.chomp
    parent_permission = parent_permission.downcase == "y"
    
    # *this line should've been placed in the 'create_person()' method
    Student.new(age, name, parent_permission)
  end
  
  # *Bad example of Single responsability rule, this is also a
  # *Bad example of YAGNI and KISS as the contents fo this method could've been placed somewhere else
  # if you see this Remind them about the Best practices and Solid lesson: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid
  def create_teacher(age, name)
    # *all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.
    print "Specialization: "
    specialization = gets.chomp
    
    # *this line should've been placed in the 'create_person()' method
    Teacher.new(age, specialization, name)
  end
end
```